java/code/src/com/redhat/rhn/domain/rhnpackage/PackageFactory.java

```diff
@@ -152,6 +175,22 @@ public static Package lookupByIdAndOrg(Long id, Org org) {
         return lookupById(id);
     }
 
+    /**
+     * Lookup Packages by the id, in the context of a given org. Does security
+     * check to verify that the org has access to the package.
+     * @param ids List of the Packages to search for
+     * @param org the org which much have access to the package
+     * @return List of Packages found
+     */
+    public static List<Package> lookupByIdAndOrg(List<Long> ids, Org org) {
+        if (!UserManager.verifyPackagesAccess(org, ids)) {
+            // User doesn't have access to the package... return null as if it
+            // doesn't exist.
+            return null;
```

Comment:
we should return `Optional` here instead of `null` for any new lookup functions.

Reply:
This is returning a collection so I would say that a ```Collections.emptyList()``` is a better fit in this case.